src/controllers/user.controller.ts

```diff
@@ -430,3 +434,89 @@ export const declineFriendRequest = async (req: Request, res: Response) => {
     return res.status(500).send({ message: error?.message });
   }
 };
+
+/**
+ * Uploads a profile image for the user with the specified userId to cloudinary
+ * and updates the user's field to point to the new image.
+ *
+ * DIMENSIONS:    automatically resizes the image to the specified dimensions
+ * USER_FIELDS:   the MongoDB fields in the user document that will be updated with the new image url
+ *                of the resized image
+ *
+ * NOTE:          the order of the dimensions and user fields must match
+ *
+ * @param req - the Request object containing the userId in the params and the image in the body
+ * @param res - the Response object sent back to the client
+ * @returns Returns either an error response with a 400 or 500 status code and a message,
+ * or a success response with a 200 status code and a message
+ */
+export const uploadProfileImg = async (req: Request, res: Response) => {
+  const DIMENSIONS = [128, 256];
+  const USER_FIELDS = ['imgUrlProfileSmall', 'imgUrlProfileLarge'];
+
+  // pass everything to multer upload so we can retrieve the image from req.file
+  upload(req, res, async err => {
+    if (err instanceof MulterError) {
+      return res.status(500).send({ message: err?.message });
+    }
+
+    const userId = req.params?.userId;
+    const image = req.file;
+
+    if (!ObjectId.isValid(userId)) {
+      return res.status(400).send({ message: 'Invalid user ID!' });
+    }
+
+    if (!image) {
+      return res.status(400).send({ message: 'No image provided!' });
+    }
+
+    try {
+      const targetUser = await User.findById(userId);
+
+      if (targetUser === null) {
+        return res.status(400).send({ message: 'User does not exist!' });
+      }
+
+      // transform image to 128x128 and 256x256 and upload to cloudinary
+      for (let i = 0; i < DIMENSIONS.length; i++) {
+        const dimensionString = `${DIMENSIONS[i]}x${DIMENSIONS[i]}`;
+
+        // use streamifier to convert buffer to stream
+        const uploadStream = cloudinary.uploader.upload_stream(
+          {
+            // the public id of the image
+            public_id: `${userId}-${dimensionString}`,
+            // the folder in cloudinary to upload to
+            folder: 'profile-images',
+            // overwrite the image if it already exists
+            overwrite: true,
+            // the transformation to apply to the image
+            transformation: [
+              {
+                radius: 'max', // crop to a circle
```

Comment:
I don't think we need store it as cropped to a circle in case we need the square image for whatever reason. We can just set a border radius in the frontend.